### DISCUSSION TITLE
Decide if we should support mixing ES6 and CommonJS modules

### DISCUSSION THREAD

**Discussion by vmutafov** [eclipse/dirigible#1159] [Ideas]
We should decide if Dirigible users would be allowed to mix **CommonJS** and **ES6 modules**. Although it seems great at a first glance to support both specifications in a single project, there may be complications while implementing this. 
If it is not hard for implementing, it would be good to support both specifications as it would allow users to easily migrate their applications to the ES6 modules syntax. We can see how NodeJS approached this and if they support both specifications to be used in a single project. Maybe we can differentiate the two specs by the file extension - `.js` for CommonJS and `.mjs` for ES6 moduled JS files.

Progress to be tracked in: https://github.com/eclipse/dirigible/issues/1147

Related Discussion: [Eclipse Dirigible 6.0](https://github.com/eclipse/dirigible/discussions/1057)

**Comment by ThuF** [comment]
@vmutafov converted to discussion, to expose the topic and also receive feedback from the community.

**Comment by ThuF** [comment]
If possible, I think it would be great to support both and to differentiate between them, based on the file extension, either `*.js` or `*.mjs`.

**Comment by delchev** [comment]
A reference from V8 documentation on why `*.mjs` extension is better to be used: https://v8.dev/features/modules#mjs

If we can "solve" with it also the irritating issue for unrecognisable client-side and server-side script files in a single project, will be very beneficial long-term as well.
